test/shared_utils/utilities.py

```diff
@@ -71,9 +71,12 @@ def compare_results_to_reference_file(
     reference_data = open_datatree(reference_file_name)
     results_data = open_datatree(results_file_name)
 
-    assert results_data.identical(reference_data), (
-        'Output and reference files ' 'do not match.'
-    )
+    try:
+        assert results_data.identical(reference_data), (
+            'Output and reference files ' 'do not match.'
+        )
+    except AssertionError as exception:
+        print_error(exception)
```

Comment:
(Same comment applies to the analogous change to `compare_results_to_reference_file_new`):

With these changes, the exception that was being raised is now being swallowed. This means that if the output file from running the request doesn't match the reference file, all that will happen is the `AssertionError` message is printed. This is good and bad:

Good: This means all tests will run, even if one of the tests fail.
Bad: The notebook will consider a run with failed tests successful, because none of the notebook cells will fail.

I think one of the basic requirements is that we need the notebook to fail if a test fails.

Reply:
There aren't emojis on here but I am face palming. Of course we need the notebook to fail, that's a pretty good point. 

> Would it be preferable to run all the tests, and then fail the notebook at the end
If this is an easy addition, *I* actually like it a lot (but of course let's get Matt's opinion on this).

Reply:
> Would it be preferable to run all the tests, and then fail the notebook at the end
If this is an easy addition, *I* actually like it a lot (but of course let's get Matt's opinion on this).

Reply:
7dea3d14c0

Reply:
I think this one is now resolved. (See other comments about failing-fast etc)